app/models/user.rb

```diff
@@ -36,4 +36,16 @@ class User < ApplicationRecord
   has_many :classrooms
   has_many :student_classrooms
   has_many :attending_classrooms, through: :student_classrooms, source: :classroom
+
+  # Returns the AR instances of students for a specific teacher
+  def students
+    User.joins(:student_classrooms).where(student_classrooms: { classroom_id: classrooms })
+  end
+
+  # Returns the AR instances of teachers for a specific student
+  def teachers
+    # Get Classroom IDs for student and retrieve the teachers who share the same classrooms
+    User.joins(:classrooms).where(classrooms: {id: student_classrooms.pluck(:classroom_id)})
```

Comment:
I'm sure you have a rubocop warning in this line, please correct it

also another possible way of writing it would be `classrooms: attending_classrooms` which I think is a bit more readable